### PR TITLE
main/xinit: fix default program execution

### PR DESCRIPTION
**Pull request by nmeum**
The exec(1) man page states the following:

> If command is specified, exec shall not return to the shell; rather
> the exit status of the process shall be the exit status of the program
> implementing command, which overlaid the shell.

Thus the shell will not continue execution after `exec startxfce4`
failed. To fix this we now check if the program exists and if that's the
case execute it using exec.

### CODE REVIEW

File: main/xinit/xinitrc [hunk 1]
```diff
@@ -45,11 +45,8 @@ if [ -d /etc/X11/xinit/xinitrc.d ] ; then
...
-# Fall back to fluxbox
-exec startfluxbox
-
-# Start only terminal as last resort
-exec xfce4-terminal || exec mrxvt || exec urxvt || exec st
+for p in startxfce4 startfluxbox xfce4-terminal \
+		mrxvt urxvt st xterm; do
+	command -v "$p" >/dev/null && exec "$p"
+done
+unset p
```
Comment (nmeum) [inline main/xinit/xinitrc]: This is never reached unless now suitable program to execute was found. So we could remove this line.

File: main/xinit/xinitrc [hunk 2]
```diff
@@ -45,11 +45,8 @@ if [ -d /etc/X11/xinit/xinitrc.d ] ; then
...
-# If not present, try the system default
-exec startxfce4
-
-# Fall back to fluxbox
-exec startfluxbox
-
-# Start only terminal as last resort
-exec xfce4-terminal || exec mrxvt || exec urxvt || exec st
+for p in startxfce4 startfluxbox xfce4-terminal \
+		mrxvt urxvt st xterm; do
```
Comment (nmeum) [inline main/xinit/xinitrc]: I also added xterm to the list of programs.

File: main/xinit/xinitrc (hunk 1 above)
Comment (ncopa) [reply main/xinit/xinitrc]: Yeah, we can remove the `unset p`.

Comment (nmeum) [reply main/xinit/xinitrc]: Done

**Comment by ncopa** [comment]
checksum  error

**Pull request closed without merging**